### PR TITLE
optimization for whisper model with decoder masked multihead attention

### PR DESCRIPTION
**Pull request by zhanghuanrong**
* graph tools update
* cuda kernel update
* operator spec update and implementation update
* greed search bug fix on wrong assumption for cross/self attention input length
* avoid use of "" name in value info when loading graph which historically in many model

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py [hunk 1]
```diff
@@ -89,13 +97,19 @@ def chain_model(args):
 
     # Initializers/opsets
     # Delete shared data between decoder/encoder and move to larger graph initializers
+    if update_decoder_subgraph_share_buffer_and_use_decoder_masked_mha(decoder_model.graph):
+        print("*****update t5 decoder subgraph successfully!!!*****")
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py:101]: >t5 [](http://example.com/codeflow?start=27&length=2)

whisper

File: onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py [hunk 2]
```diff
@@ -89,13 +97,19 @@ def chain_model(args):
...
+    else:
+        print("*****DecoderMaskedMultiHeadAttention is not applied to T5 decoder*****")
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py:103]: >T5 [](http://example.com/codeflow?start=70&length=2)

whisper

Comment (zhanghuanrong) [reply onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py:96]: fix it later as conceptually t5.

File: onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py (hunk 1 above)
Comment (zhanghuanrong) [reply onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py:94]: fix it later as conceptually t5.

File: onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py [hunk 3]
```diff
@@ -82,13 +90,19 @@ def chain_model(args):
 
     # Initializers/opsets
     # Delete shared data between decoder/encoder and move to larger graph initializers
+    if update_decoder_subgraph_share_buffer_and_use_decoder_masked_mha(decoder_model.graph):
```
Comment (yufenglee) [inline onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py:93]: It should be called only for GPU

File: onnxruntime/core/graph/contrib_ops/bert_defs.cc [hunk 4]
```diff
@@ -676,6 +676,11 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
                "which beam the 'k' th token came from for the 'j' th beam for batch 'i' in the current iteration",
                "M",
                OpSchema::Optional)
+        .Input(10,
+               "bias",
+               "Bias tensor with shape (hidden_size + hidden_size + v_hidden_size) from input projection",
+               "T",
+               OpSchema::Optional)
```
Comment (yufenglee) [inline onnxruntime/core/graph/contrib_ops/bert_defs.cc:683]: you need to update ContribOperators.md and OperatorKernels.md to pass CI.

File: onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py (hunk 3 above)
Comment (zhanghuanrong) [reply onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py]: yeah, fixed

File: onnxruntime/core/graph/contrib_ops/bert_defs.cc (hunk 4 above)
Comment (zhanghuanrong) [reply onnxruntime/core/graph/contrib_ops/bert_defs.cc:683]: fixed

**Pull request merged**